src/main/java/com/networknt/schema/BaseJsonValidator.java

```diff
@@ -137,67 +142,83 @@ && noExplicitDiscriminatorKeyOverride(discriminatorMapping, jsonSchema)) {
-            // this is where A -> B -> C inheritance exists, A has the root discriminator and B adds to the mapping
+            // this is where A -> B -> C inheritance exists, A has the root discriminator
+            // and B adds to the mapping
             final JsonNode propertyName = discriminatorOnSchema.get("propertyName");
             if (null != propertyName) {
-                throw new JsonSchemaException(instanceLocation + " schema " + schema + " attempts redefining the discriminator property");
+                throw new JsonSchemaException(
+                        instanceLocation + " schema " + schema + " attempts redefining the discriminator property");
             }
             final ObjectNode mappingOnContextDiscriminator = (ObjectNode) discriminator.get("mapping");
             final ObjectNode mappingOnCurrentSchemaDiscriminator = (ObjectNode) discriminatorOnSchema.get("mapping");
             if (null == mappingOnContextDiscriminator && null != mappingOnCurrentSchemaDiscriminator) {
-                // here we have a mapping on a nested discriminator and none on the root discriminator, so we can simply
+                // here we have a mapping on a nested discriminator and none on the root
+                // discriminator, so we can simply
                 // make it the root's
                 discriminator.set("mapping", discriminatorOnSchema);
             } else if (null != mappingOnContextDiscriminator && null != mappingOnCurrentSchemaDiscriminator) {
-                // here we have to merge. The spec doesn't specify anything on this, but here we don't accept redefinition of
+                // here we have to merge. The spec doesn't specify anything on this, but here we
+                // don't accept redefinition of
                 // mappings that already exist
                 final Iterator<Map.Entry<String, JsonNode>> fieldsToAdd = mappingOnCurrentSchemaDiscriminator.fields();
-                while (fieldsToAdd.hasNext()) {
-                    final Map.Entry<String, JsonNode> fieldToAdd = fieldsToAdd.next();
-                    final String mappingKeyToAdd = fieldToAdd.getKey();
-                    final JsonNode mappingValueToAdd = fieldToAdd.getValue();
-
-                    final JsonNode currentMappingValue = mappingOnContextDiscriminator.get(mappingKeyToAdd);
-                    if (null != currentMappingValue && currentMappingValue != mappingValueToAdd) {
-                        throw new JsonSchemaException(instanceLocation + "discriminator mapping redefinition from " + mappingKeyToAdd
-                                + "/" + currentMappingValue + " to " + mappingValueToAdd);
-                    } else if (null == currentMappingValue) {
-                        mappingOnContextDiscriminator.set(mappingKeyToAdd, mappingValueToAdd);
-                    }
-                }
+                addFieldsToMappingWithCheck(fieldsToAdd, mappingOnContextDiscriminator, instanceLocation);
```

Comment:
This refactor seems rather arbitrary to pull the method out, as the method is private and thus doesn't add any benefit of testability.